plugins/notifications-backend/src/service/router.ts

```diff
@@ -49,62 +147,89 @@ export async function createRouter(
+        {},
+        c.request.query,
+      );
+      // we need to convert strings to real types due to open PR https://github.com/openapistack/openapi-backend/pull/571
+      q.pageNumber = stringToNumber(q.pageNumber);
+      q.pageSize = stringToNumber(q.pageSize);
+      q.read = stringToBool(q.read);
 
-    getNotifications(dbClient, catalogClient, q, q.pageSize, q.pageNumber, q)
-      .then(notifications => res.json(notifications))
-      .catch(next);
-  });
+      getNotifications(
+        dbClient,
+        loggedInUser,
+        catalogClient,
+        q,
+        q.pageSize,
+        q.pageNumber,
+        q,
+      )
+        .then(notifications => res.json(notifications))
+        .catch(next);
+    },
+  );
 
-  api.register('getNotificationsCount', (c, _, res: express.Response, next) => {
-    const q: Paths.GetNotificationsCount.QueryParameters = Object.assign(
-      {},
-      c.request.query,
-    );
+  api.register(
+    'getNotificationsCount',
+    async (c, req: express.Request, res: express.Response, next) => {
+      const loggedInUser = await getLoggedInUser(req);
+      await checkPermission(req, notificationsReadPermission, loggedInUser);
 
-    // we need to convert strings to real types due to open PR https://github.com/openapistack/openapi-backend/pull/571
-    q.read = q.read = stringToBool(q.read);
+      const q: Paths.GetNotificationsCount.QueryParameters = Object.assign(
+        {},
+        c.request.query,
+      );
 
-    getNotificationsCount(dbClient, catalogClient, q)
-      .then(result => res.json(result))
-      .catch(next);
-  });
+      // we need to convert strings to real types due to open PR https://github.com/openapistack/openapi-backend/pull/571
+      q.read = q.read = stringToBool(q.read);
+
+      getNotificationsCount(dbClient, loggedInUser, catalogClient, q)
+        .then(result => res.json(result))
+        .catch(next);
+    },
+  );
 
-  api.register('setRead', (c, _, res: express.Response, next) => {
-    const messageId = c.request.query.messageId.toString();
-    const user = c.request.query.user.toString();
-    const read = c.request.query.read.toString() === 'true';
+  api.register(
+    'setRead',
+    async (c, req: express.Request, res: express.Response, next) => {
+      const loggedInUser = await getLoggedInUser(req);
+      await checkPermission(req, notificationsSetReadPermission, loggedInUser);
 
-    setRead(dbClient, messageId, user, read)
-      .then(result => res.json(result))
-      .catch(next);
-  });
+      const messageId = c.request.query.messageId.toString();
+      const read = c.request.query.read.toString() === 'true';
 
-  // create router
+      setRead(dbClient, loggedInUser, messageId, read)
+        .then(result => res.json(result))
+        .catch(next);
+    },
+  );
 
+  // create router
   const router = Router();
   router.use(express.json());
+  router.use(permissionIntegrationRouter);
```

Comment:
what does this do? i don't see we use it somewhere.

Reply:
I think it's a leftover when playing with conditional authorization which we probably don't need. I will test and potentially remove that.